electron/main.ts

```diff
@@ -100,23 +100,33 @@ app.on('activate', () => {
   }
 });
 
-function openFile() {
-  dialog
-    .showOpenDialog({
-      properties: ['openFile', 'multiSelections'],
-      filters: [{ name: 'Maps', extensions: ['json', 'geojson'] }]
-    })
-    .then((data) => {
-      if (!data) return;
-
-      const { filePaths } = data;
-      // TODO: Change the fact that implementation currently only supports opening one file at a time.
-
-      fs.readFile(filePaths[0], 'utf-8', (err, data) => {
-        // We could also probably just send an alert/dialogue saying something went wrong
-        if (err) return;
+async function openFolder() {
+  const fileChannel = 'file-content';
+  try {
+    const data = await dialog.showOpenDialog({
+      properties: ['openDirectory']
+    });
+    const { filePaths } = data;
+    console.log(`filePath`, filePaths[0]);
+    let promises: Promise<any>[] = [];
 
-        win?.webContents.send('file-content', JSON.parse(data.toString()));
-      });
+    const files = await fs.readdir(filePaths[0]);
+    files.forEach((file: string) => {
```

Comment:
This can be a `map` call instead of pushing to a `promises` array.
```ts
const promises = files.map((file: string) => {
  // ...
  return fs.readFile(...);
}
```